### PR TITLE
handle error where react-on-rails package entry missing

### PR DESCRIPTION
**Pull request by benjiwheeler**
This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/602)

### CODE REVIEW

**Comment by justin808** [comment]
Nice work @benjiwheeler! Could you add a simple rspec test? I'll merge right after that.

**Comment by justin808** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/602)*

**Comment by justin808** [comment]
@benjiwheeler you have lint errors: https://travis-ci.org/shakacode/react_on_rails/builds/174902848

try 

cd client
npm run lint -- --fix

Offenses:
lib/react_on_rails/version_checker.rb:62:36: C: Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
        if parsed_package_contents.has_key?("dependencies") &&
                                   ^^^^^^^^
lib/react_on_rails/version_checker.rb:63:10: C: Style/MultilineOperationIndentation: Align the operands of a condition in an if statement spanning multiple lines.
         parsed_package_contents["dependencies"].has_key?("react-on-rails")
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
lib/react_on_rails/version_checker.rb:63:50: C: Style/PreferredHashMethods: Use Hash#key? instead of Hash#has_key?.
         parsed_package_contents["dependencies"].has_key?("react-on-rails")
                                                 ^^^^^^^^
93 files inspected, 3 offenses detected
rake aborted!
Command failed with status (1): [cd /home/travis/build/shakacode/react_on_r...]
/home/travis/build/shakacode/react_on_rails/rakelib/task_helpers.rb:20:in `block in sh_in_dir'
/home/travis/build/shakacode/react_on_rails/rakelib/task_helpers.rb:20:in `each'
/home/travis/build/shakacode/react_on_rails/rakelib/task_helpers.rb:20:in `sh_in_dir'
/home/travis/build/shakacode/react_on_rails/rakelib/lint.rake:7:in `block (2 levels) in <top (required)>'
/home/travis/.rvm/gems/ruby-2.3.1/gems/rake-11.2.2/exe/rake:27:in `<top (required)>'
/home/travis/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `eval'
/home/travis/.rvm/gems/ruby-2.3.1/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => default => lint => lint:lint => lint:rubocop
(See full trace by running task with --trace)

**Comment by benjiwheeler** [comment]
Having trouble running rspec with this Gem bundle... getting:

`Your bundle is locked to activejob (4.2.3), but that version could not be found in any of the sources listed in your Gemfile. If you haven't changed sources, that means the author of activejob (4.2.3) has removed it.`

**Comment by justin808** [comment]
@benjiwheeler Did you try running bundle install? you might need to cd spec/dummy and run bundle install there too!

See https://github.com/shakacode/react_on_rails/blob/master/CONTRIBUTING.md

**Comment by benjiwheeler** [comment]
yes, ran bundle install on app dir and spec/dummy both... tried changing ruby version and rerunning... now getting mini_racer error... reinstalling xcode commandline utils etc...

**Comment by justin808** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[.delete_by, line 1 at r2](https://reviewable.io:443/reviews/shakacode/react_on_rails/602#-KWur9EGzDKMN3M9xSz3:-KWur9EGzDKMN3M9xSz4:b-142obk) ([raw file](https://github.com/shakacode/react_on_rails/blob/8f770159d80ed735e11cabe313b779bae34f5c45/.delete_by#L1)):*
> ```
> 20170101
> ```

what is this file?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/602)*

**Comment by justin808** [comment]
@benjiwheeler Let's get this in the next release coming up soon!

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/602#-:-KWurMYL1Qo5po24JH7-:b-3y36s4)*

File: .delete_by [hunk 1]
```diff
@@ -0,0 +1 @@
+20170101
```
Comment (justin808) [inline .delete_by]: @benjiwheeler Can you please remove this file from the PR.

**Comment by benjiwheeler** [comment]
removed .delete_by

**Comment by justin808** [comment]
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/602)*

**Comment by justin808** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/602#-:-KYI9awi5j-PDlYn6Lid:bnfp4nl)*

**Pull request merged**